toqito/matrices/pauli.py

```diff
@@ -1,12 +1,12 @@
 """Pauli matrices."""
 
 import numpy as np
-from scipy import sparse
+from scipy.sparse import csr_array
 
 from toqito.matrix_ops import tensor
 
 
-def pauli(ind: int | str | list[int] | list[str], is_sparse: bool = False) -> np.ndarray | sparse.csr_matrix:
+def pauli(ind: int | str | list[int] | list[str], is_sparse: bool = False) -> csr_array:
```

Comment:
Why remove `np.ndarray`? You get a `csr_array` for the output only when we provide `is_sparse` is not `False`.

Reply:
I was thinking back to: https://github.com/vprusso/toqito/pull/521#discussion_r1543072900 and that's why initially I removed `is_sparse` and made some changes to the test cases. But I'll wait for a further response.

Reply:
Yes, but `issparse` is a method for both sparse array and sparse matrix types. So, the structure of the function shouldn't change that much. Before, the return type was an array or a sparse matrix. Now, it can be an `np.ndarray` for all options of allowed inputs. 

https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.issparse.html#scipy.sparse.issparse